azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub.py

```diff
@@ -40,34 +35,34 @@ def _execute_op(self, op):
             self.auth_provider.on_sas_token_updated_handler = CallableWeakMethod(
                 self, "on_sas_token_updated"
             )
-            operation_flow.delegate_to_different_op(
-                stage=self,
-                original_op=op,
-                new_op=pipeline_ops_iothub.SetIoTHubConnectionArgsOperation(
+            self._send_worker_op_down(
+                worker_op=pipeline_ops_iothub.SetIoTHubConnectionArgsOperation(
                     device_id=self.auth_provider.device_id,
                     module_id=getattr(self.auth_provider, "module_id", None),
                     hostname=self.auth_provider.hostname,
                     gateway_hostname=getattr(self.auth_provider, "gateway_hostname", None),
                     ca_cert=getattr(self.auth_provider, "ca_cert", None),
                     sas_token=self.auth_provider.get_current_sas_token(),
+                    callback=op.callback,
```

Comment:
>callback=op.callback, [](start = 20, length = 21)

This is kind of confusing, because this callback you're setting immediately gets overridden in the implementation of the `_send_worker_op_down` method.